rollup/internal/controller/sender/sender.go

```diff
@@ -446,7 +397,7 @@ func (s *Sender) resubmitTransaction(feeData *FeeData, auth *bind.TransactOpts,
 
 // checkPendingTransaction checks the confirmation status of pending transactions against the latest confirmed block number.
 // If a transaction hasn't been confirmed after a certain number of blocks, it will be resubmitted with an increased gas price.
-func (s *Sender) checkPendingTransaction(header *types.Header, confirmed uint64) {
+func (s *Sender) checkPendingTransaction(header *gethTypes.Header, confirmed uint64) {
 	number := header.Number.Uint64()
 	atomic.StoreUint64(&s.blockNumber, number)
```

Comment:
If deployed replicas, I think this number is not right.

if receipt on A, but insert on B, the inserted block number will be wrong.

Reply:
Yes. `submit_block_number` is not the block number for which the transaction is included. It only reflects the roughly submit time so that the sender can decide whether to resubmit or not.

Reply:
If the node's sync heights of each replica are the same (or up to the latest block), it's OK. And if not, there would be some problem triggering the resubmit logic.

Reply:
fixed in [f22b884](https://github.com/scroll-tech/scroll/pull/1076/commits/f22b884667637d4fd164dca65ea6df6a12c71d76)